plasmapy/physics/dispersion/two_fluid.py

```diff
@@ -0,0 +1,127 @@
+    """
+
+    pi_angle = theta * (np.pi / 180.0)
+    sine = np.sin(pi_angle)
+    cosine = np.cos(pi_angle)
+    tangent = sine / cosine
+    cs = np.sqrt(beta / 2.0) * speed
+    di = 1
+    caksq = (cosine ** 2) * (speed ** 2)
+    cmsq = (speed ** 2) + (cs ** 2)
+    D = 1 + wavenumber ** 2 * de2
+
+    term1 = -(speed ** 2 / D + cs ** 2 + caksq * (1 + wavenumber ** 2 * di ** 2 / D) / D) * wavenumber ** 2
+    term2 = caksq * wavenumber ** 4 * (speed ** 2 / D + cs ** 2 + cs ** 2 * (1 + wavenumber ** 2 * di ** 2 / D)) / D
+    term3 = -(cs ** 2 * wavenumber ** 6 * caksq ** 2) / D ** 2
+
+    dispersion_relation = (1, term1, term2, term3)
+
+    roots = np.roots(dispersion_relation)
+    sq_roots = np.sqrt(roots)
+
+    phase_speeds = sq_roots / wavenumber
+    return sq_roots, phase_speeds
+
+
+def single_theta_value(beta=0.6, ca=1., de2=0.000545, theta=0., kmin=1e-2, kmax=10., npoints=200):
+    """
+    Compute the dispersion relation w(k) vs k for a single theta value.
+
+    beta: Total plasma beta,
+    ca: Alfven speed based on mean field,
+    de2: me/mi,
+    theta: Angle of propagation in degrees
+    kkmin: Minimum Wavenumber of interest in units of kdi
+    kkmax: Maximum wavenumber of interest in units of kdi
+
+    Output is an array with 4 columns, k, w-fast, w-alf, w-slow
+    """
+
+    kmmn = np.log10(kmin)
+    kmmx = np.log10(kmax)
+    wavenumber = np.logspace(kmmn, kmmx, npoints)
+    warray = np.zeros((3, npoints))
+    for i in range(0, npoints):
+        f, s = tfps(beta, ca, de2, theta, wavenumber[i])
+        warray[:, i] = f
+    plt.loglog(wavenumber, warray[0, :], label='Fast/Magnetosonic')
+    plt.loglog(wavenumber, warray[1, :], label='Alfven/KAW')
+    plt.loglog(wavenumber, warray[2, :], label='Slow')
+    plt.xlabel('$kd_i$')
+    plt.ylabel('$\omega/\omega_{ci}$')
+    plt.legend(loc='best', fancybox=True, framealpha=0.2)
+    plt.title('Dispersion Relation for beta=' + str(beta) + ' and me/mi=' + str(de2))
+    plt.show()
+
+
+class TFEV:
+    def __init__(self, beta=0.6, ca=1., de2=0.000545, theta=0., k=1., aa=0.1):
+        """
+        beta: Total plasma beta,
+        ca: Alfven speed based on mean field,
+        de2: me/mi,
+        theta: Angle of propagation in degrees
+        k: wavenumber of interest in units of kdi
+
+        Output: Prints the two fluid eigenvector to the screen
+        """
+
+        self.beta = beta
+        self.ca = ca
+        self.de2 = de2
+        self.theta = theta
+        self.k = k
+        self.aa = aa
+        self.f, _ = tfps(self.beta, ca, de2, theta, k)
+
+    def amp(self, w):
+        th = self.theta * np.pi / 180.
+        bb = 1 - w ** 2 * (1 + self.de2 * self.k ** 2) / (self.k ** 2 * np.cos(th) ** 2)
+        sk = 'sin(' + str(round(self.k, 3)) + 'x)'
+        ck = 'cos(' + str(round(self.k, 3)) + 'x)'
+
+        def st(a):
+            return str(round(a, 3))
+
+        return {'bx': 0.,
```

Comment:
I would put this in a separate line, define the dict with a descriptive variable name and then return said variable.

Reply:
Or rather, does this need to be a dict? Could you just setup the class, run the dispersion solver, and then make each of these values in the dict a separate property/method?